### PR TITLE
Updates to release scripts

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Contains the following changes:

* Commit `pom.xml` after updating Substrate to release version
* Global configuration for GIT
* Use `set -e` to fail fast

### CODE REVIEW

File: .ci/update-archetypes.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 
-export ARCHETYPE_REPO_SLUG=gluonhq/client-maven-archetypes
-export XML_LOCATION=src/main/resources/META-INF/maven/archetype-metadata.xml
+ARCHETYPE_REPO_SLUG=gluonhq/client-maven-archetypes
```
Comment (tiainen) [inline .ci/update-archetypes.sh]: `export` is not required?

Comment (abhinayagarwal) [reply .ci/update-archetypes.sh]: We do not want to set env variables. We just want to use them in the script.

File: .ci/update-archetypes.sh [hunk 2]
```diff
@@ -16,5 +16,5 @@ for f in ./client-archetype-* ; do
   fi
 done
 
-git -c user.name="Gluon Bot" -c user.email="githubbot@gluonhq.com" commit */$XML_LOCATION -m "Upgrade client-maven-plugin version to $1"
+git commit */$XML_LOCATION -m "Update client-maven-plugin version to $1"
```
Comment (tiainen) [inline .ci/update-archetypes.sh]: I assume the github `user.name` and `user.email` variables are set globally and are persisted during the entire travis job. These scripts (`update-archetypes.sh` and `update-samples.sh`) are never called apart from `release.sh`?

Comment (abhinayagarwal) [reply .ci/update-archetypes.sh]: No, they are not. Updating samples and archetypes are a part of the release process and only called from release script.

**Pull request merged**